### ISSUE TITLE
_NET_WM_STATE_ABOVE support for floating windows

### ISSUE THREAD

**Issue by lilydjwg** [i3/i3#4265]
## I'm submitting a…

<pre>
[ ] Bug
[x] Feature Request
[ ] Documentation Request
[ ] Other (Please describe in detail)
</pre>

## Current Behavior
There is no way to make a floating window above other floating windows.

## Desired Behavior

Support _NET_WM_STATE_ABOVE and an `above toggle|enable|disable` command.

## Environment

Output of `i3 --moreversion 2>&-`:
<pre>
Binary i3 version:  4.19 (2020-11-15) © 2009 Michael Stapelberg and contributors
Running i3 version: 4.19 (2020-11-15) (pid 92941)
Loaded i3 config: /home/lilydjwg/.config/i3/config (Last modified: 2020年11月22日 星期日 17时42分58秒, 477 seconds ago)

The i3 binary you just called: /usr/bin/i3
The i3 binary you are running: i3
</pre>

<pre>
- Linux Distribution & Version: Arch Linux latest
- Are you using a compositor (e.g., xcompmgr or compton): picom
</pre>

**Comment by Airblader** [comment]
In i3 all floating windows are above all tiled windows, which essentially already behaves like _NET_NET_WM_STATE_ABOVE:

> _NET_WM_STATE_ABOVE indicates that the window should be on top of most windows

Given that i3 is a tiling window manager having a bunch of floating windows on top of each other just sounds like not using i3 the intended way, and we don't really want to add too much complexity to the floating layer by coordinating different floating windows among each other. Therefore I'm going to close this issue.

**Comment by lilydjwg** [comment]
> Given that i3 is a tiling window manager having a bunch of floating windows on top of each other just sounds like not using i3 the intended way

No, but it's GIMP.

I really like the way i3 is configured, its windows structured, its IPC mechanism. But today I gave my first try to i3 and sadly I find many features missing compared to Awesome.

**Comment by Airblader** [comment]
GIMP has two modes, with all detached windows or as one. At least this used to be the case last time I used it, I figure it's still like that.

**Comment by wlhlm** [comment]
It's in the menubar under "Windows" > "Single-Window Mode".

**Comment by lilydjwg** [comment]
Yes GIMP has two modes. With the single window mode you just get one single window with tabs. It's not easy to compare things with overlapping windows whose certain regions are carefully aligned.

**Comment by orestisfl** [comment]
Re-opening to consider supporting `_NET_WM_STATE_ABOVE`.

**Comment by brunetton** [comment]
> Re-opening to consider supporting `_NET_WM_STATE_ABOVE`.

Can't wait ! 
Here is another use-case:

I'm using [Free Poker DB](https://github.com/jejellyroll-fr/fpdb-3) that opens a lot of floating windows that follows a poker table (runing in Wine) and have to stay on top of them. Something like that:

![](https://github.com/jejellyroll-fr/fpdb-3/wiki/Images/hud.webp)

But I didn't found any solution in i3 to make those HUD (Heads Up Display) windows stay on top of the poker table when the poker table is in floating mode (or in a container that is floating). It only works when the window, or a parent container is not floating. And it makes sense as floatings are on top of non floatings.

The big problem is that some poker softwares do not support window resizing, they are not made for this and 100% unusable when resized (many labels are on top of each other). 
If I want to use the HUD (in top of this software):
- I can't use the poker software in floating mode as HUD will be hidden behind
- if make the poker software tiled, my HUD will be on top but I can't use the software
    - the workaround I'm using for now is to create an useless window (like a software that displays an image) and put it to the bottom of the poker software, make them horizontally stacked and try to adjust the proportions of the window to make it more or less displayed correctly; I mean being globally usable)

Being able to add `above enable` for this software class in my i3-config file would solve my problem 100% !

**Comment by brunetton** [comment]
Another usecase: I'm also using Reaper quite a lot, like described in https://github.com/i3/i3/issues/5617#issuecomment-1670225555 and have the same kind of problem. `above` option would save my life here too.

**Comment by sbradnick** [comment]
`kitty` requires `_NET_WM_STATE_ABOVE` for the `quick-access-terminal` [kitten](https://sw.kovidgoyal.net/kitty/kittens/quick-access-terminal/):
```
$ kitten quick-access-terminal
[0.153] [glfw error 65544]: X11: Window manager does not support _NET_WM_STATE_ABOVE

...

$ echo $XDG_CURRENT_DESKTOP 
i3

$ wmctrl -m
Name: i3
Class: N/A
PID: N/A
Window manager's "showing the desktop" mode: N/A
```

**Comment by classabbyamp** [comment]
I'm also interested in `_NET_WMA_STATE_ABOVE` for kitty's quick access terminal
